walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,62 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Set;
+
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<String> meetingAttendees = request.getAttendees();
+    long meetingDuration = request.getDuration();
+    int dayStart = TimeRange.WHOLE_DAY.start();
+    int dayEnd = TimeRange.WHOLE_DAY.end();
+    boolean[] unavailableMinutes = new boolean[dayEnd - dayStart];
```

Comment:
Small nitpick: for readability, it's usually better to have positive booleans. Negative booleans are harder to understand, e.g. `!unavailableMinutes[i]` is more confusing than `availableMinutes[i]`.

Reply:
I agree. I did so because an initialised boolean array has a default value of false, so when iterating through the events array to block off the minutes that were unavailable, I had to convert them from false to true, and so a 'true' value in the unavailableMinutes array to indicate a busy minute made sense, although this leads to many double negatives.

I just realised that there is an Arrays.fill() function that could convert the whole boolean array from false to true, so this solved the problem!